source/ast/expressions/MiscExpressions.cpp

```diff
@@ -986,6 +986,8 @@ Expression& AssertionInstanceExpression::fromLookup(const Symbol& symbol,
 
     ASTContext bodyContext(*symbolScope, LookupLocation::max);
     bodyContext.assertionInstance = &instance;
+    // Propagate previously founded time advance specs
+    bodyContext.flags |= ASTFlags::PropertyTimeAdvance;
```

Comment:
This should probably only set the flag if the parent context had the flag set right?

Reply:
fixed